### ISSUE TITLE
Enhancement: [restrict-template-expressions] Restrict optional chaining in template strings

### ISSUE THREAD

**Issue by chris-orgorg** [typescript-eslint/typescript-eslint#10709]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [x] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [x] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/restrict-template-expressions/

### Description

It would be great to add restriction of optional chaining in template strings.  For example:

`${obj?.prop}` would short circuit to 'undefined' which is not what most developers would intend.

### Fail

```TypeScript
`${obj?.prop}`
```

### Pass

```TypeScript
`${obj.prop}`
```

**Comment by Josh-Cena** [comment]
We already have a rule in this spirit, which does more than what's asked here: https://typescript-eslint.io/rules/restrict-template-expressions. If you want a rule that's purely syntax-based, then it doesn't pertain to TypeScript specifically and therefore wouldn't be met by our inclusion criteria.

**Comment by chris-orgorg** [comment]
Ah thanks, I thought allowNullish was defaulted to false
